integration/e2emimir/client.go

```diff
@@ -820,12 +820,16 @@ func (c *Client) DeleteRuleGroup(namespace string, groupName string) error {
 	defer cancel()
 
 	// Execute HTTP request
-	res, err := c.httpClient.Do(req.WithContext(ctx))
+	resp, err := c.httpClient.Do(req.WithContext(ctx))
 	if err != nil {
 		return err
 	}
 
-	defer res.Body.Close()
```

Comment:
> // On error, any Response can be ignored. A non-nil Response with a
// non-nil error only occurs when CheckRedirect fails, and even then
// the returned Response.Body is already closed.

I think this was moved by accident when I introduced the status code check; In https://github.com/grafana/mimir/pull/8444/commits/49b7719a03258d9c452df691b6dab15ac94d6409, I have moved the body closure  before the status the status code  check - I linked the text above because I'm not sure what you meant with "even in case of an error". An error from `Do` or the status code check so I double checked the `Do` documentation.

Sorry if the above was obvious, it wasn't for me hence why I had to read the docs.